src/core/buffers/types.ts

```diff
@@ -75,11 +75,20 @@ export interface IBufferStateActive {
 // State emitted when the buffer has been filled to the end
 export interface IBufferStateFull {
   type : "full-buffer";
-  value : {
-    bufferType : IBufferType; // The type of the Representation
-  };
+
+  // The type of the Representation
+  value : { bufferType : IBufferType };
 }
 
+export interface IProtectedSegmentEvent {
+  type : "protected-segment";
+  value : { type : string;
+            data : Uint8Array;
+            content: { adaptation : Adaptation;
+                       manifest : Manifest;
+                       period : Period;
+                       representation : Representation; }; }; }
```

Comment:
As said before:

Depending on our solution, we might not need to add the content there.